src/js/media-controller.js

```diff
@@ -90,14 +90,24 @@ class MediaController extends MediaContainer {
       // This current assumes that the media controller is the fullscreen element
       // which may be true in most cases but not all.
       // The prior version of media-chrome supported alt fullscreen elements
-      // and that's something we can work towards here
+      // and that's something we can work towards here.
+      //
+      // Generally the fullscreen and PiP API's have the exit methods and enabled
+      // flags on the `document`. The active element is accessed on the document
+      // or shadow root. Entering fullscreen or PiP is called on the element. i.e.
+      //
+      //   - Document.exitFullscreen()
+      //   - Document.fullscreenEnabled
+      //   - Document.fullscreenElement / ShadowRoot.fullscreenElement
+      //   - Element.requestFullscreen()
+      //
       MEDIA_ENTER_FULLSCREEN_REQUEST: () => {
         const docOrRoot = this.getRootNode();
         const media = this.media;
 
         if (docOrRoot.pictureInPictureElement) {
```

Comment:
Ok so understanding which el is pip needs the shadow root when in the shadow dom but exiting always requires the document. Confusing.